lib/colorado_lottery.rb

```diff
@@ -0,0 +1,40 @@
+class ColoradoLottery
+
+  attr_reader :registered_contestants,
+              :winners,
+              :current_contestants,
+              :name
+
+  def initialize
+    @registered_contestants = {}
+    @winners = []
+    @current_contestants = {}
+  end
+
+  def interested_and_18?(contestant, game_name)
+    if contestant.game_interests != game_name.name
+      return false
+    elsif contestant.game_interests == game_name.name && contestant.age >= 18
+      true
+    end
+  end
+
+  def can_register?(contestant, game_name)
+    if contestant.age >= 18 || contestant.game_interests == game_name.name
+      return true
+    end
+  end
+
+  def register_contestant(contestant, game_name)
+    @registered_contestants[game_name] = contestant
```

Comment:
close, but not quite-- the key is set correctly, but the value needs to be an _array_ of contestants.  And setting the value equal doesn't work when you have more than one contestant-- it will always reset to a single contestant. so you'd want to be setting an empty array if the key has never been set before, and then shoveling into the array. 

```
  def register_contestant(contestant, game)
    if registered_contestants[game.name] == nil
      registered_contestants[game.name] = []
    end
    registered_contestants[game.name] << contestant
end
```